services/core-api/app/api/now_applications/resources/administrative_amendment_list_resource.py

```diff
@@ -0,0 +1,180 @@
+from flask_restplus import Resource, inputs
+from flask import request, current_app
+from sqlalchemy_filters import apply_pagination, apply_sort
+from sqlalchemy import desc, func, or_, and_
+from werkzeug.exceptions import BadRequest
+from datetime import datetime, timezone
+
+from app.extensions import api, db
+from app.api.mines.mine.models.mine import Mine
+from app.api.mines.permits.permit.models.permit import Permit
+from app.api.mines.permits.permit_amendment.models.permit_amendment import PermitAmendment
+from app.api.mines.permits.permit_amendment.models.permit_amendment_document import PermitAmendmentDocument
+from app.api.now_applications.models.now_application import NOWApplication
+from app.api.now_applications.models.now_application_identity import NOWApplicationIdentity
+from app.api.now_applications.models.now_party_appointment import NOWPartyAppointment
+from app.api.mines.documents.models.mine_document import MineDocument
+from app.api.now_applications.models.now_application_document_xref import NOWApplicationDocumentXref
+from app.api.now_applications.models.administrative_amendments.application_reason_code_xref import ApplicationReasonXref
+from app.api.now_applications.response_models import NOW_APPLICATION_MODEL
+from app.api.now_applications.models.applications_view import ApplicationsView
+from app.api.utils.access_decorators import requires_role_edit_permit, requires_any_of, VIEW_ALL, GIS
+from app.api.utils.resources_mixins import UserMixin
+from app.api.utils.custom_reqparser import CustomReqparser
+
+
+def map_notice_of_work_type_from_permit_number(permit_number):
+    first_charecter = permit_number[0]
+    return {'P': 'PLA', 'G': 'SAG', 'M': 'MIN', 'C': 'COL', 'Q': 'QCA'}.get(first_charecter, None)
+
+
+class AdministrativeAmendmentListResource(Resource, UserMixin):
+    parser = CustomReqparser()
+    parser.add_argument('mine_guid', type=str, required=True)
+    parser.add_argument('received_date', type=str, required=True)
+    parser.add_argument('permit_amendment_guid', type=str, required=True)
+    parser.add_argument('permit_id', type=int, required=True)
+    parser.add_argument('application_source_type_code', type=str, required=True)
+    parser.add_argument('application_reason_codes', type=list, location='json', required=True)
+
+    @api.doc(description='Adds a Notice of Work to a mine/permit.', params={})
```

Comment:
The description needs to be updated for AA. And what about the marshal_with model name?

Reply:
AA and NOWs using the same model as they are the same from the entity perspective, so no need to change marshaling